making-mods/advanced-js/mutation-observers.md

```diff
@@ -0,0 +1,199 @@
+# Mutation Observers
+
+Author(s): ortiza5 (nomadic)
+
+There are many different elements to Vivaldi's interface that can be toggled on and off with different settings or keyboard shortcuts. While this is great for customizability, JavaScript mods that affect these UI elements can easily be inadvertently broken by simple actions such as hiding a toolbar, entering fullscreen, or switching over to the built in email experiment.
+
+You can always tell the users that they aren't allowed to use various settings that you know are incompatible with your mod, but in order to make a more robust mod that fits with a wider range of users with diverse browser workflows, you should make your mod handle changes in the UI gracefully. One way to accomplish this is with Mutation Observers.
+
+## Contents
+- [Basics of mutation observers](#basics-of-mutation-observers)
+  - [Target element](#target-element-targetelement)
+  - [Configuration options](#configuration-options-configurationoptions)
+  - [Handler function for mutations](#handler-function-for-mutations-callback)
+- [Code examples](#code-examples)
+  - [Fullscreen observer](#fullscreen-observer)
+  - [Address bar / mail bar observer](#address-bar--mail-bar-observer)
+  - [Toggling UI element (status bar) observer](#toggling-ui-element-status-bar-observer)
+
+## Basics of mutation observers
+
+The main purpose of Mutation Observers is, as the name suggests, to observe for mutations. These mutations, or changes, can be for things like adding or removing an element and changing attributes like classes on an element.
+
+The basic structure of a Mutation Observer should look something like this:
+
+```JavaScript
+// Fill in "..." with appropriate values
+
+const targetElement = document.querySelector(" ... ");
+const configurationOptions = { ... : true};
+
+function callBack (mutations) {
```

Comment:
`callback` could be considered one word, so I guess that can be changed. The space afterwards must have been a mistake. I messed with my auto-formatter, so I might have messed that up.

Fixed in 7a81981

Anonymous functions should have the space. That is considered more standard practice from what I can find. Here is an excerpt from [here ](https://www.crockford.com/code.html) explaining the reasoning:
> "If a function literal is anonymous, there should be one space between the word function and the ( left parenthesis. If the space is omitted, then it can appear that the function's name is function, which is an incorrect reading."

--

> And thinking of it, `() => {}`, or `function() {}`?

I think we shouldn't worry about this too much. There are valid reasons to switch between the two depending on the usage of `this` wanted, so I am fine with letting people choose as they see fit and not specifying one over the other in CONTRIBUTING. What do you think?

Reply:
Oh, okay. Both your arguments make sense.